src/cluster/placement/config.go

```diff
@@ -69,6 +71,30 @@ func (c *Configuration) NewOptions() Options {
 	return opts
 }
 
+// DeepCopy makes a deep copy of the configuration.
+func (c Configuration) DeepCopy() (Configuration, error) {
+	b, err := yaml.Marshal(c)
+	if err != nil {
+		return Configuration{}, err
+	}
+	var res Configuration
+	if err := yaml.Unmarshal(b, res); err != nil {
+		return Configuration{}, err
+	}
+	return res, nil
+}
+
+// ApplyOverride applys the override values.
+func (c Configuration) ApplyOverride(opts *placementpb.Options) Configuration {
+	if opts == nil {
+		return c
+	}
+	if opts.IsSharded != nil {
+		c.IsSharded = &opts.IsSharded.Value
```

Comment:
Instead of taking an address to ptr value, perhaps better to copy the value?  Otherwise the ownership semantics is strange (mutate the original override and it mutates the configuration struct).

e.g.

```go
isShardedValueCopy := *opts.IsSharded.Value
c.IsSharded = &isShardedValueCopy
```